projects/canopy/src/lib/forms/checkbox-group/checkbox-group.component.spec.ts

```diff
@@ -0,0 +1,218 @@
+
+  beforeEach(async(() => {
+    errorStateMatcherMock = mock(LgErrorStateMatcher);
+
+    TestBed.configureTestingModule({
+      imports: [FormsModule, ReactiveFormsModule],
+      declarations: [
+        TestCheckboxGroupComponent,
+        LgCheckboxGroupComponent,
+        LgToggleComponent,
+        MockComponents(LgValidationComponent, LgHintComponent, LgIconComponent),
+      ],
+      providers: [
+        {
+          provide: LgErrorStateMatcher,
+          useFactory: () => instance(errorStateMatcherMock),
+        },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestCheckboxGroupComponent);
+    component = fixture.componentInstance;
+
+    groupDebugElement = fixture.debugElement.query(
+      By.directive(LgCheckboxGroupComponent),
+    );
+    groupInstance = groupDebugElement.injector.get<LgCheckboxGroupComponent>(
+      LgCheckboxGroupComponent,
+    );
+
+    hintDebugElement = fixture.debugElement.query(By.directive(LgHintComponent));
+
+    fieldsetDebugElement = fixture.debugElement.query(By.css('fieldset'));
+
+    checkboxDebugElements = fixture.debugElement.queryAll(By.css('lg-toggle'));
+
+    checkboxInstances = checkboxDebugElements.map(debugEl => debugEl.componentInstance);
+
+    fixture.detectChanges();
+  }));
+
+  it('sets all checkbox buttons to the same name', () => {
+    expect(groupInstance.name.length > 0).toBe(true);
+    const name = checkboxInstances.pop().name;
+    for (const checkbox of checkboxInstances) {
+      expect(checkbox.name).toBe(name);
+    }
+  });
+
+  it('checks the selected checkbox button when a value is provided', () => {
+    groupInstance.value = ['red'];
+    fixture.detectChanges();
+    const checkedOption: DebugElement = checkboxDebugElements.find(
+      checkboxDebugElement => checkboxDebugElement.componentInstance.checked === true,
+    );
+    expect(checkedOption.componentInstance.value).toContain('red');
+  });
+
+  it('sets unique ids on all the checkbox buttons', () => {
+    const checkboxIds = checkboxInstances.map(({ id }) => id);
+    expect(new Set(checkboxIds).size).toBe(checkboxIds.length);
+    for (const id of checkboxIds) {
+      expect(/lg-toggle-\d{1,3}/.test(id)).toBe(true);
+    }
+  });
+
+  it('marks the selected checkbox when the value is changed', () => {
+    const redOption: DebugElement = checkboxDebugElements.find(
+      checkboxDebugElement => checkboxDebugElement.componentInstance.value === 'red',
+    );
+    redOption.query(By.css('input')).triggerEventHandler('click', null);
+    fixture.detectChanges();
+    const selected = checkboxInstances.find(checkbox => checkbox.value === 'red');
+    expect(selected.checked).toBe(true);
+    const notSelected = checkboxInstances.filter(checkbox => checkbox.value !== 'red');
+    for (const checkbox of notSelected) {
+      expect(checkbox.checked).toBe(false);
+    }
+  });
+
+  it('updates the model value when a checkbox option is checked', () => {
```

Comment:
I wonder if we should add another couple of tests to check that:
- when multiple checkboxes are selected, it adds them to the array
- when selected checkboxes are unselected, it removes them
I think would feel a little more like a complete set of unit tests, what do you think?

Reply:
It makes sense having tests to check those things too. Will add them.

Reply:
Done.